chore: proving broker benchmarks

danielntmd opened the pull request:
Proving Broker Benchmarks:
- Enqueueing jobs tested against various proof sizes respective to the size of the epoch and epoch number.
- Dequeueing jobs tested against various proof sizes respective to the size of the epoch
- Mock database implementation for fast benchmarking setup (if we want to minimize database operations)
- Initialization and cleanup of prover broker from various proof sizes respective to the size of the epoch.

Commit:
WIP: Add proving broker benchmarks and mock agent

Commit:
reverting accidental changes

Commit:
chore: add proving broker benchmarks
-add enqueue benchmarks for min, small, medium, large, and max epoch proofs
-add enqueue benchmarks for random and interleaved epochs
-add dequeue benchmarks for min, small, medium, large, and max epoch proofs
-add mock database implementation for fast benchmarking
-add benchmark results to githuba actions
-remove mock agent
-rename benchmarking file

mralj reviewed (approved):
Generally looks good to me 👍 
I have left a couple of nitpicky comments.
Once this is ready, please ask @PhilWindle  for review as he has much more knowledge and context here than I, and, he'll be able to do a proper review :)

PhilWindle reviewed (changes requested):
Looks like a great start, thanks so much for this!

Commit:
chore: add changes from review and cleanup
-add logger to replace console logging
-add dynamic mapping to the proving broker's priority order (no hardcoded proof order)
-remove mock inputs and replace with a mock proof store that can return GCP-like URIs
-add more metrics to collect ie. median, P90, P95, P99, Jobs/s
-add concurrency to both enqueue (eliminates database batching stalls) and dequeue functions (simulates multiple agents)
-add KVBrokerDatabase (with default configs) as the main database (with cleanup) and remove the in-line memory database
-fix the epoch proofs and validate that they adhere to the block and epoch rules
-change maximum epoch to be 32 blocks with 400 txs
-add seeded rng generator for consistent random epoch generation across builds
-clean job building into separate function
-clean mock proof store
-add export tag to proving broker's priority list

Commit:
chore: rm mock database

PhilWindle committed:
Merge branch 'next' into proving_broker_benchmarks

Commit:
chore: add proving broker initialization benchmarks and cleanup benchmarks on empty and full epochs
-add database counting helper function

-add initialization benchmarks for proving broker to start from a loaded kvbrokerdatabase
-add cleanup benchmarks for no deletion for an epoch size
-add assertions to make sure benchmarks are correct
-add formatting output to githubactions

Commit:
Merge branch 'proving_broker_benchmarks' of github.com:danielntmd/aztec-packages into proving_broker_benchmarks

Commit:
chore: add assertion to verify cleanup, adjust p95 to be <= 95, and clean up the enqueue metrics

danielntmd committed:
Merge branch 'next' into proving_broker_benchmarks

danielntmd committed:
Merge branch 'next' into proving_broker_benchmarks

PhilWindle committed:
Merge branch 'next' into proving_broker_benchmarks

Commit:
chore(fix): rename to give more resources during CI

Commit:
chore(fix): move benchmark into prover-client root test directory to enable docker isolation when running CI

danielntmd committed:
Merge branch 'next' into proving_broker_benchmarks

Commit:
chore: disable logging to reduce redis stress

Commit:
chore: add prover broker benchmark output to bench-out directory

Commit:
chore(fix): isolate and use resources for prover broker benchmarks

alexghr commented on test_enum_serialization.js:
I think this can be removed?

Commit:
Revert "chore(fix): isolate and use resources for prover broker benchmarks"

This reverts commit 60c676c29d5ed62f0f9686b1b1f9e8f2992b6154.

Commit:
chore(fix): revert and isolate and grant resources to benchmark
